src/wasi.zig

```diff
@@ -651,9 +651,48 @@ pub fn integrationTest() void {
         return;
     }
 
+    if (!testReadfile()) {
+        return;
+    }
+
     log.fatal.print("Integration test passed\n");
 }
 
+fn testReadfile() bool {
+    @setRuntimeSafety(false);
+
+    // path_open
+    const fd_addr_in_linear_memory = 0;
+    const path = "test.txt";
+    const path_addr_in_linear_memory = 4;
+    @memcpy(@as([*]u8, @ptrFromInt(linear_memory_offset + path_addr_in_linear_memory)), path);
+    var res = path_open(3, 0, path_addr_in_linear_memory, path.len, 0, 0, 0, 0, fd_addr_in_linear_memory);
+    if (@intFromEnum(res) != 0) {
+        log.fatal.printf("path_open failed: res={d}\n", .{@intFromEnum(res)});
+        return false;
+    }
+    const open_fd = @as(*i32, @ptrFromInt(fd_addr_in_linear_memory + linear_memory_offset));
+
+    // fd_read
+    var buf_iovec_ptr = @as(*IoVec, @ptrFromInt(path_addr_in_linear_memory + 20));
+    buf_iovec_ptr.buf = 32;
+    buf_iovec_ptr.buf_len = 1024;
+    res = fd_read(open_fd.*, @as(i32, @intCast(path_addr_in_linear_memory + 20)), 1, @intCast(path_addr_in_linear_memory + 8 + 20));
```

Comment:
The argument of `size_addr` is the size of the content, not the size of the buffer.
This is set by `fd_read`. (https://github.com/mewz-project/mewz/blob/main/src/wasi.zig#L143)

So we need to pass other region of memory, for example `2048`.

Reply:
Du har ret, det bruges til at gemme, hvor mange tegn indholdet optager.